mongo/options/clientoptions.go

```diff
@@ -1166,7 +1167,19 @@ func addClientCertFromSeparateFiles(cfg *tls.Config, keyFile, certFile, keyPassw
 		return "", err
 	}
 
-	data := make([]byte, 0, len(keyData)+len(certData)+1)
+	keySize := len(keyData)
+	if keySize > 64*1024*1024 {
+		return "", errors.New("X.509 key must be less than 64 MiB")
+	}
+	certSize := len(certData)
+	if certSize > 64*1024*1024 {
+		return "", errors.New("X.509 certificate must be less than 64 MiB")
+	}
+	dataSize := keySize + certSize + 1
+	if dataSize > math.MaxInt {
+		return "", errors.New("size overflow")
+	}
+	data := make([]byte, 0, dataSize)
```

Comment:
I'm surprised this check satisfies the CodeQL rule because `dataSize > math.MaxInt` can never be `true` (`dataSize` is an `int`, so it can never be greater than the max `int` value). Nevertheless, it only hurts code clarity minimally.

Optional: Consider a check that won't overflow if it's greater than the max int value:
```suggestion
	dataSize := int64(keySize) + int64(certSize) + 1
	if dataSize > math.MaxInt {
		return "", errors.New("size overflow")
	}
	data := make([]byte, 0, int(dataSize))
```